# Filtering DE_Tdu_Tpr_for_Tml
# Input Tdu_Tpr_voom_Tml/DE_Tdu_Tpr_min10_3rep.txt
# 5,137 orthologs in file bayes_flag_sig_Tml_for_UR.csv; these orthologs showed non-biased mapping and were analyzed for homeolog-specific expression analysis (not all 5,400 orthologs showing unbaised mapping have homeolog-specific expression profiles)

In [1]:
import pandas as pd
import sys
import os
import glob

In [2]:
os.chdir("/ufrc/soltis/shan158538/TragFL_NewAnalysis/OutPut/DE_analysis/Tdu_Tpr_voom_Tml/")

In [3]:
df = pd.read_table("DE_Tdu_Tpr_min10_3rep.txt")

In [4]:
df.head()

logFC    ...            B
Tpr_TRINITY_DN11257_c2_g1|Tdu_TRINITY_DN16696_c... -0.395358    ...    -4.847267
Tpr_TRINITY_DN10844_c2_g7|Tdu_TRINITY_DN25328_c... -0.596919    ...    -2.714478
Tpr_TRINITY_DN15383_c4_g16|Tdu_TRINITY_DN14160_...  1.009462    ...    -2.534116
Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c3_g1 -0.586770    ...    -3.231621
Tpr_TRINITY_DN10853_c1_g4|Tdu_TRINITY_DN21860_c... -0.232925    ...    -6.216757

[5 rows x 6 columns]

In [5]:
df.shape
# shows the number of rows and columns, separately

(8248, 6)

In [6]:
unbias = pd.read_csv("bayes_flag_sig_Tml_for_UR.csv", sep = ',')

In [7]:
unbias.head()

commonID          ...           flag_sig_Tml_tdu_tpr
0  Tpr_TRINITY_DN11281_c1_g9|Tdu_TRINITY_DN15992_...          ...                              0
1  Tpr_TRINITY_DN11283_c3_g1|Tdu_TRINITY_DN25596_...          ...                              0
2  Tpr_TRINITY_DN11283_c4_g7|Tdu_TRINITY_DN21302_...          ...                              0
3  Tpr_TRINITY_DN11284_c2_g19|Tdu_TRINITY_DN20652...          ...                              0
4  Tpr_TRINITY_DN11285_c1_g4|Tdu_TRINITY_DN17239_...          ...                              1

[5 rows x 14 columns]

In [8]:
unbias.shape

(5137, 14)

In [9]:
keys = list(unbias.commonID)
# creating a list containing all elements from the column commonID

In [10]:
len(keys)
# count the number of elements in the list

5137

In [11]:
keys[:10]
# show the first few elements of the list

['Tpr_TRINITY_DN11281_c1_g9|Tdu_TRINITY_DN15992_c0_g1',
 'Tpr_TRINITY_DN11283_c3_g1|Tdu_TRINITY_DN25596_c4_g2',
 'Tpr_TRINITY_DN11283_c4_g7|Tdu_TRINITY_DN21302_c2_g2',
 'Tpr_TRINITY_DN11284_c2_g19|Tdu_TRINITY_DN20652_c0_g3',
 'Tpr_TRINITY_DN11285_c1_g4|Tdu_TRINITY_DN17239_c2_g3',
 'Tpr_TRINITY_DN11294_c1_g1|Tdu_TRINITY_DN19017_c2_g10',
 'Tpr_TRINITY_DN11295_c2_g2|Tdu_TRINITY_DN23322_c3_g3',
 'Tpr_TRINITY_DN11295_c2_g3|Tdu_TRINITY_DN12723_c0_g1',
 'Tpr_TRINITY_DN11296_c5_g1|Tdu_TRINITY_DN20745_c2_g4',
 'Tpr_TRINITY_DN11296_c6_g2|Tdu_TRINITY_DN25140_c1_g1']

In [12]:
df_filtered = df[df.index.isin(keys)]
# remain the indexes (orthologs names) in df that are shown in the list keys; output is df_filtered

In [13]:
df_filtered.head()

logFC    ...            B
Tpr_TRINITY_DN10844_c2_g7|Tdu_TRINITY_DN25328_c... -0.596919    ...    -2.714478
Tpr_TRINITY_DN15383_c4_g16|Tdu_TRINITY_DN14160_...  1.009462    ...    -2.534116
Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c3_g1 -0.586770    ...    -3.231621
Tpr_TRINITY_DN15495_c0_g2|Tdu_TRINITY_DN17403_c...  0.424043    ...    -5.739230
Tpr_TRINITY_DN11223_c1_g1|Tdu_TRINITY_DN21812_c...  0.390412    ...    -6.240813

[5 rows x 6 columns]

In [14]:
df_filtered.shape

(3817, 6)

# Among the 8,248 orthologs with DE data, 3,817 showed unbiased mapping and have homeolog-specific expression profiles

In [15]:
df_filtered.to_csv("DE_Tdu_Tpr_filtered.txt")

# Output file: DE_Tdu_Tpr_filtered.txt

In [16]:
Filtered = pd.read_table("DE_Tdu_Tpr_filtered.txt", sep = ",")

In [17]:
Filtered.shape

(3817, 7)

In [18]:
Filtered.head()

Unnamed: 0    ...            B
0  Tpr_TRINITY_DN10844_c2_g7|Tdu_TRINITY_DN25328_...    ...    -2.714478
1  Tpr_TRINITY_DN15383_c4_g16|Tdu_TRINITY_DN14160...    ...    -2.534116
2  Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c...    ...    -3.231621
3  Tpr_TRINITY_DN15495_c0_g2|Tdu_TRINITY_DN17403_...    ...    -5.739230
4  Tpr_TRINITY_DN11223_c1_g1|Tdu_TRINITY_DN21812_...    ...    -6.240813

[5 rows x 7 columns]

In [19]:
Equal = Filtered[Filtered["adj.P.Val"] >= 0.05]

In [20]:
Equal.shape

(3162, 7)

# adj.P.Val >= 0.5 means there is no expression difference between Tdu and Tpr; there are 3,162 of them

In [21]:
Equal.head()

Unnamed: 0    ...            B
3  Tpr_TRINITY_DN15495_c0_g2|Tdu_TRINITY_DN17403_...    ...    -5.739230
4  Tpr_TRINITY_DN11223_c1_g1|Tdu_TRINITY_DN21812_...    ...    -6.240813
5  Tpr_TRINITY_DN14551_c1_g11|Tdu_TRINITY_DN21735...    ...    -4.527947
6  Tpr_TRINITY_DN13298_c1_g1|Tdu_TRINITY_DN18431_...    ...    -5.366188
7  Tpr_TRINITY_DN11712_c1_g8|Tdu_TRINITY_DN19492_...    ...    -7.407547

[5 rows x 7 columns]

# adj.P.Val < 0.5 means the expression level is significantly different between Tdu and Tpr
# If logFC is larger than 0, Tdu > Tpr; if logFC is smaller than 0, Tdu < Tpr
# 278 orthologs have higher expression in Tdu; 377 orthologs expressed higher in Tpr

In [22]:
Tdu_higher = Filtered[(Filtered["adj.P.Val"] < 0.05) & (Filtered["logFC"] > 0)]

In [23]:
Tdu_higher.shape

(278, 7)

In [24]:
Tdu_higher.head()

Unnamed: 0    ...            B
1   Tpr_TRINITY_DN15383_c4_g16|Tdu_TRINITY_DN14160...    ...    -2.534116
13  Tpr_TRINITY_DN10334_c5_g3|Tdu_TRINITY_DN16861_...    ...     0.132213
17  Tpr_TRINITY_DN10516_c1_g3|Tdu_TRINITY_DN19561_...    ...    -2.908788
28  Tpr_TRINITY_DN12437_c6_g1|Tdu_TRINITY_DN18152_...    ...    -1.862780
65  Tpr_TRINITY_DN12896_c9_g2|Tdu_TRINITY_DN20213_...    ...     0.140104

[5 rows x 7 columns]

In [25]:
Tpr_higher = Filtered[(Filtered["adj.P.Val"] < 0.05) & (Filtered["logFC"] < 0)]

In [26]:
Tpr_higher.shape

(377, 7)

In [27]:
Tpr_higher.head()

Unnamed: 0    ...            B
0   Tpr_TRINITY_DN10844_c2_g7|Tdu_TRINITY_DN25328_...    ...    -2.714478
2   Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c...    ...    -3.231621
19  Tpr_TRINITY_DN11462_c1_g3|Tdu_TRINITY_DN18955_...    ...     2.597909
20  Tpr_TRINITY_DN14614_c1_g6|Tdu_TRINITY_DN1401_c...    ...    -2.878470
29  Tpr_TRINITY_DN12134_c0_g2|Tdu_TRINITY_DN21624_...    ...    -2.940983

[5 rows x 7 columns]

In [28]:
Equal.to_csv("DE_Tdu_Tpr_filtered_not_sig_loci.txt", index=False)
Tdu_higher.to_csv("DE_Tdu_Tpr_filtered_sig_loci_Tdu_higher.txt", index=False)
Tpr_higher.to_csv("DE_Tdu_Tpr_filtered_sig_loci_Tpr_higher.txt", index=False)